specs/lightclient/validator.md

```diff
@@ -0,0 +1,395 @@
+        SyncCommitteeSignature(
+            slot=state.slot,
+            index=index,
+            signature=signature,
+        ) for index in sync_committee_indices
+    ]
+```
+
+##### Broadcast sync committee signature
+
+The validator broadcasts the assembled signature(s) to the assigned subnet(s), the `sync_committee_{subnet_id}` pubsub topic.
+
+The `subnet_id` is derived from the position in the sync committee such that the sync committee is divided into "subcommittees". It can be computed via `compute_subnet_for_sync_committee`:
+
+```python
+def compute_subnet_for_sync_committee(sync_committee_index: SyncCommitteeIndex) -> uint64:
+    return uint64(sync_committee_index % SYNC_COMMITTEE_SUBNET_COUNT)
+```
+
+*NOTE*: subnet assignment does not change during the duration of a validator's assignment to a given sync committee.
+
+*NOTE*: (TODO) revisit subnet algo -- may be a bit more efficient to make "contiguous" "subcommittees"
+*NOTE*: (TODO) details of when to join subnets
+
+#### Sync committee contributions
+
+Each slot some sync committee members are selected to aggregate the `SyncCommitteeSignature`s into a `SyncCommitteeContribution` which is broadcast on a global channel for inclusion into the next block.
+
+##### Aggregation selection
+
+A validator is selected to aggregate based on the computation in `is_sync_committee_aggregator` where `state` is a `BeaconState` with the relevant sync committees for the queried `slot`, i.e. `state.slot` is within the span covered by the current or next sync committee period. `privkey` is the BLS private key associated with the validator.
+
+```python
+def get_sync_committee_slot_signature(state: BeaconState, slot: Slot, privkey: int) -> BLSSignature:
+    domain = get_domain(
+      state, 
+      DOMAIN_SYNC_COMMITTEE_SELECTION_PROOF, 
+      compute_epoch_at_slot(slot)
+    )
+    signing_root = compute_signing_root(slot, domain)
+    return bls.Sign(privkey, signing_root)
+
+def is_sync_committee_aggregator(state: BeaconState, slot: Slot, privkey: int) -> bool:
+    signature = get_sync_committee_slot_signature(state, slot, privkey)
+    modulo = max(
+      1, 
+      SYNC_COMMITTEE_SIZE // TARGET_AGGREGATORS_PER_SYNC_COMMITTEE
+    )
+    return bytes_to_uint64(hash(signature)[0:8]) % modulo == 0
+```
+
+*NOTE*: the set of aggregators will generally change every slot; however, the assignments can be computed ahead of time as soon as the committee is known
+
+##### Construct sync committee contribution
+
+If a validator is selected to aggregate the `SyncCommitteeSignature`s produced on a subnet during a given `slot`, they construct an aggregated `SyncCommitteeContribution`. Given all of the collected `sync_committee_signatures` from the `sync_committee_{subnet_id}` gossip, create the contribution via `construct_sync_committee_contribution`. `state` in this function is the same `BeaconState` used by the aggregator to produce their individual signature for the assigned slot.
+
```

Comment:
Slight preference for more granular setting of fields as was done in [validator guide aggregation](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/phase0/validator.md#construct-aggregate)

not a big deal either way but a bit easier to parse I think